### PR TITLE
feat: Create Queue Implementation using Linked List

### PR DESCRIPTION
**Pull request by saumyasingh048**
#### Description of Change
I have provided algorithm for implementation of queue using Linked List.

#### Checklist

- [ ] Added description of change
- [ ] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#File-Name-guidelines)
- [ ] Added tests and example, test must pass
- [ ] Relevant documentation/comments is changed or added
- [ ] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [ ] Search previous suggestions before making a new one, as yours may be a duplicate.
- [ ] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C/pull/720"></a>

### CODE REVIEW

File: data_structures/linked_list/Queue Implementation using Linked List [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    }
+}
+int search(Q *p, int item)
+{
+    queue *q;
+    q = p->front;
+    while(q != NULL)
+    {
+        if(q->data == item)
+            return q;
```
Comment (hemanth-kotagiri) [inline data_structures/linked_list/Queue Implementation using Linked List:73]: There is a potential warning in your code.
```In function ‘search’:
warning: returning ‘queue *’ from a function with return type ‘int’ makes integer from pointer without a cast [-Wint-conversion]
   73 |             return q;
```
Please look into it.

**Review by hemanth-kotagiri — CHANGES_REQUESTED** [review]
Please add the ``.c`` file extension also for the file. It is missing. Take note of the warnings that I have commented earlier as well.

**Comment by saumyasingh048** [comment]
I’ve made the changes and made the pull request once again . Kindly look into it now .


Sent from Yahoo Mail for iPhone


On Monday, October 26, 2020, 7:24 PM, Hemanth Kotagiri <notifications@github.com> wrote:




@hemanth-kotagiri requested changes on this pull request.

Please add the .c file extension also for the file. It is missing. Take note of the warnings that I have commented earlier as well.

In data_structures/linked_list/Queue Implementation using Linked List:
> +    queue *q;
+    q = p->front;
+    while(q != NULL)
+    {
+        printf("%d\n", q->data);
+        q = q->next;
+    }
+}
+int search(Q *p, int item)
+{
+    queue *q;
+    q = p->front;
+    while(q != NULL)
+    {
+        if(q->data == item)
+            return q;

There is a potential warning in your code.
warning: returning ‘queue *’ from a function with return type ‘int’ makes integer from pointer without a cast [-Wint-conversion]
   73 |             return q;

Please look into it.

File: data_structures/linked_list/QueueLinkedList.c [hunk 2]
```diff
@@ -0,0 +1,133 @@
+#include<stdio.h>
+#include<stdlib.h>
+
+typedef struct queue
+{
+    int data;
+    struct queue *next;
+}queue;
```
Comment (hemanth-kotagiri) [inline data_structures/linked_list/QueueLinkedList.c:8]: Please maintain spacing between chunks of code. It is important to have the code well structured.

File: data_structures/linked_list/QueueLinkedList.c [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+{
+    queue *q;
+    q = p->front;
+    while(q != NULL)
+    {
+        if(q->data == item)
+            return q;
+        q = q->next;
+    }
+    printf("Element NOT Found.");
```
Comment (hemanth-kotagiri) [inline data_structures/linked_list/QueueLinkedList.c:76]: The non-void function that you have written here doesn't always return a value in all control paths. Please fix this issue.

File: data_structures/linked_list/QueueLinkedList.c [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+            else
+                printf("UNDERFLOW");
+
+            break;
+        case 4:
+            printf("Enter element to search: ");
+            scanf("%d", &item);
+            if(!isEmpty(&q))
+            {
+                queue item_loc = search(&q, item);
```
Comment (hemanth-kotagiri) [inline data_structures/linked_list/QueueLinkedList.c:123]: There is an error in your code here. Please fix it.
```In function ‘main’:
error: invalid initializer
Line 123 |    queue item_loc = search(&q, item);
```

File: data_structures/linked_list/QueueLinkedList.c [hunk 5]
```diff
@@ -0,0 +1,133 @@
+#include<stdio.h>
```
Comment (hemanth-kotagiri) [inline data_structures/linked_list/QueueLinkedList.c:1]: My humble request is that you please compile the code that you write before pushing it to this PR. I am finding compile-time errors/warnings which you can fix beforehand.

**Review by hemanth-kotagiri — CHANGES_REQUESTED** [review]
Here are a few suggestions @saumyasingh048 .
1. **Always compile** the code that you write and check for errors/warnings and **fix them beforehand**.
2. Please **maintain spacing** between functions/chunks of code and **follow the official style guide**.
3. Please remove the redundant file of which I have left a comment on.

**Comment by hemanth-kotagiri** [comment]
@saumyasingh048, if you would like to stick to the PR: #724 , please close this to avoid redundant PRs.

**Comment by hemanth-kotagiri** [comment]
@Panquesito7 Can you please close this PR because another PR from the same user is raised: #724

**Comment by Panquesito7** [comment]
> close this PR

I'll close the other PR; this one is older than #724.

**Comment by hemanth-kotagiri** [comment]
@Panquesito7 

Also, are you open for collaboration? I would love to join to review PRs, issues and contribute! 

Thanks!

**Comment by Panquesito7** [comment]
> would love to join to review PRs, issues and contribute

Keep contributing, project maintainers will later decide if you are capable to join the team. 🙂
For now, you can do the following in the [C-Plus-Plus](https://github.com/TheAlgorithms/C-Plus-Plus) and [C](https://github.com/TheAlgorithms/C) repositories:

- Review pull requests often.
- Watch the repository to get latest notifications.
- Fix errors or improve code/documentation via PR's.
- And much more you can do to help.

If you have any other questions, you can
[contact me](https://github.com/Panquesito7#contact) or other project maintainers.

Good to know you're interested in contributing! Thank you. 😃

**Review by hemanth-kotagiri — COMMENTED** [review]
@Panquesito7, you can remove this redundant PR as well adding the labels ``duplicate``, ``invalid`` as you did for #724 .

Thanks!

**Pull request closed without merging**